src/indices.jl

```diff
@@ -48,45 +49,49 @@ end
 
 Base.lastindex(s::Indices) = s.packed[end]
 
-function pageid_offset(s::Indices, i)
-    pageid = div(i - 1, INT_PER_PAGE) + 1
-    return (id = pageid, offset = (i - 1) & (INT_PER_PAGE - 1) + 1)
+@inline function pageid_offset(i)
+    t1 = i - 1
+    pageid = div(t1, INT_PER_PAGE)
+    t2 = t1 & INT_PER_PAGE_1
```

Comment:
The `&` assumes `INT_PER_PAGE` is a power of two? I guess that's always practically true, but might be nice to assert it somewhere.  Or just use `divrem` I guess — I checked and the `@code_native` of `divrem(x,y)` looks really good when the `y` is a known constant power of two.

Reply:
Chose best of both worlds :smile:.